src/main/java/uk/gov/pay/adminusers/service/InviteService.java

```diff
@@ -131,7 +134,21 @@ public void generateOtp(InviteOtpRequest inviteOtpRequest) {
                     });
             LOGGER.info("New 2FA token generated for invite code [{}]", inviteOtpRequest.getCode());
         } else {
-            throw notFoundException();
+            throw notFoundInviteException(inviteOtpRequest.getCode());
+        }
+    }
+
+    @Transactional
+    public User createInvitedUser(InviteValidateOtpRequest inviteValidateOtpRequest) {
+        Optional<InviteEntity> inviteOptional = inviteDao.findByCode(inviteValidateOtpRequest.getCode());
+        if (inviteOptional.isPresent() && secondFactorAuthenticator.authorize(inviteOptional.get().getOtpKey(), inviteValidateOtpRequest.getOtpCode())) {
```

Comment:
How you going identify between if the invite was not found or the second factor authentication error? As discussed selfservice needs to know the differentiation in order to direct user either to re-enter otp or stop registration altogether.